src/main/java/grumpygordon/parser/Parser.java

```diff
@@ -0,0 +1,304 @@
+package grumpygordon.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.util.regex.Pattern;
+
+import grumpygordon.commands.ByeCommand;
+import grumpygordon.commands.Command;
+import grumpygordon.commands.DeadlineCommand;
+import grumpygordon.commands.DeleteCommand;
+import grumpygordon.commands.EventCommand;
+import grumpygordon.commands.FindCommand;
+import grumpygordon.commands.ListCommand;
+import grumpygordon.commands.MarkCommand;
+import grumpygordon.commands.TodoCommand;
+import grumpygordon.commands.UnmarkCommand;
+import grumpygordon.exceptions.GrumpyGordonDateTimeFormatException;
+import grumpygordon.exceptions.GrumpyGordonException;
+import grumpygordon.exceptions.GrumpyGordonInitialisationException;
+import grumpygordon.exceptions.GrumpyGordonInvalidCommandException;
+import grumpygordon.tasks.Deadline;
+import grumpygordon.tasks.Event;
+import grumpygordon.tasks.Task;
+import grumpygordon.tasks.TaskList;
+import grumpygordon.tasks.Todo;
+
+/**
+ * Represents a parser that parses user input.
+ */
+public class Parser {
+    /**
+     * Separator for saved format.
+     */
+    private static final String SAVED_FORMAT_SEPARATOR_REGEX = " \\| ";
+
+    /**
+     * Separator for user input.
+     */
+    private static final String SPACE_REGEX = " ";
+    /**
+     * Regex for todo command.
+     */
+    private static final String TODO_REGEX = "todo\\s.*";
+    /**
+     * Regex for mark command.
+     */
+    private static final String MARK_REGEX = "mark\\s([0-9]+)$";
+    /**
+     * Regex for unmark command.
+     */
+    private static final String UNMARK_REGEX = "unmark\\s([0-9]+)$";
+    /**
+     * Regex for delete command.
+     */
+    private static final String DELETE_REGEX = "delete\\s([0-9]+)$";
+
+    /**
+     * Regex for find command.
+     */
+    private static final String FIND_REGEX = "find\\s.*$";
+
+    /**
+     * Regex for deadline command.
+     */
+    private static final String DEADLINE_INFO_REGEX = ".*\\s+/by\\s+.*";
+    /**
+     * Regex for event command.
+     */
+    private static final String EVENT_INFO_REGEX = ".*\\s+/from\\s+.*\\s+/to\\s+.*";
+    /**
+     * Regex for datetime.
+     */
+    private static final String DATETIME_REGEX = "\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]) \\d{2}:\\d{2}";
```

Comment:
I like how you declared this here, making it easier for you to parse the inputString later.